core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramParser.java

```diff
@@ -52,6 +51,8 @@ protected DateHistogramAggregationBuilder createFactory(String aggregationName,
             factory.interval((Long) interval);
         } else if (interval instanceof DateHistogramInterval) {
             factory.dateHistogramInterval((DateHistogramInterval) interval);
+        } else {
+            throw new AssertionError();
```

Comment:
We seem to be using this exception in a number of places already. I like that this is an error and not an exception as it means there is an error in the code rather than on the user end, and should not be caught?